src/main/java/com/datalogics/pdf/samples/forms/FillForm.java

```diff
@@ -127,38 +90,10 @@ public static void main(final String... args) throws Exception {
      * @param outputUrl The file to which the filled form will be saved
      * @throws Exception a general exception was thrown
      */
-    public static void fillPdfForm(final URL inputFormUrl, final URL inputDataUrl, final String formType,
-                                   final URL outputUrl)
+    public static void fillPdfForm(final URL inputFormUrl, final URL inputDataUrl,
+                                   final FormImporter.FormType formType, final URL outputUrl)
                     throws Exception {
-        final PDFDocument pdfDocument = DocumentUtils.openPdfDocument(inputFormUrl);
-
-        // There are two types of forms that we can fill, so find out which kind we have here.
-        final PDFDocumentType documentType = XFAService.getDocumentType(pdfDocument);
-
-        if (documentType == PDFDocumentType.Acroform) {
-            // If this is an Acroform, make sure the form data is either FDF for XFDF.
-            if (FDF_FORMAT.equalsIgnoreCase(formType)) {
-                fillAcroformFdf(pdfDocument, inputDataUrl, outputUrl);
-            } else if (XFDF_FORMAT.equalsIgnoreCase(formType)) {
-                fillAcroformXfdf(pdfDocument, inputDataUrl, outputUrl);
-            } else {
-                throw new IllegalArgumentException("Invalid formData type for Acroform document. "
-                                                   + "FDF and XFDF supported.");
-            }
-        } else if (documentType.isXFA()) {
-            // If the document has an XFA form, make sure that we were passed an XML data file.
-            // Note that PDF Java Toolkit doesn't support generating appearances or running calculations on XFA forms
-            // (though field formatting is supported), so be sure to use Acrobat or another full-featured PDF viewer
-            // to verify the output. A viewer like OSX's Preview won't cut it.
-            if (XML_FORMAT.equalsIgnoreCase(formType)) {
-                fillXfa(pdfDocument, inputDataUrl, outputUrl);
-            } else {
-                throw new IllegalArgumentException("Invalid formData type for XFA document. XML supported.");
-            }
-        } else {
-            // If the form document is not XFA or Acroform, it's either not a form or it's something we don't support.
-            throw new IllegalArgumentException("PDF File does not contain an AcroForm or an XFA form.");
-        }
+        FormImporter.fillPdfForm(inputFormUrl, inputDataUrl, formType, outputUrl);
```

Comment:
Thanks for bringing that up. For compatibility, when I moved the bulk of the public methods to ``FormImporter``, I left "forwarding" methods behind, so that they could still be called on the original class.

Eclipse offers to do this when moving methods, but I had to do this manually in IDEA.